src/main/java/org/rpgl/subevent/GetEvents.java

```diff
@@ -48,18 +53,26 @@ public void prepare(RPGLContext context) throws Exception {
      * Adds an RPGLEvent datapack ID to the subevent, to be later granted to target.
      *
      * @param eventId an RPGLEvent datapack ID
+     * @param originItem a UUID for a RPGLItem if the event was provided by an item, or null otherwise
      */
-    public void addEvent(String eventId) {
-        this.getEvents().addString(eventId);
+    public void addEvent(String eventId, String originItem) {
+        this.json.getJsonArray("events").addJsonObject(RPGLFactory.newEvent(eventId, originItem));
```

Comment:
simplify to getEvents()....?

Reply:
Nope, that method returns a different data type that wouldn't let this work (List<Object> vs List<RPGLEvent>)